### PR TITLE
Better support for multiple applications

### PR DESCRIPTION
**Pull request by shirey**
- name globus group json files starting with the first 8 digits of application id
 - name process user info files starting with the first 8 digits of application id
 - remove ability to pass in globus groups file
 - remove ability to flag sennet specific groups info
 - removed globus_groups.py, moved function get_globus_groups_info into AuthHelper
 - added ability to flag for read or write group membership in secure decorator
 - removed deprecated object worker files

### CODE REVIEW

**Comment by yuanzhou** [comment]
Test the submodule by pointing to this branch: https://github.com/dbmi-pitt/search-adaptor/pull/9

**Comment by yuanzhou** [comment]
Changed base branch to `dev-integrate` to be consistent with the related PRs for DEV integration test: 
- https://github.com/hubmapconsortium/uuid-api/pull/109
- https://github.com/hubmapconsortium/entity-api/pull/388

File: hubmap_commons/hm_auth.py [hunk 1]
```diff
@@ -14,30 +14,33 @@
 from hubmap_commons import file_helper
 from hubmap_commons import hubmap_const
 from hubmap_commons import exceptions
+from pickle import TRUE
```
Comment (yuanzhou) [inline hubmap_commons/hm_auth.py:17]: We are not using this import `from pickle import TRUE`, don't know why it was introduced.

File: hubmap_commons/activity.py [hunk 2]
```diff
@@ -4,14 +4,7 @@
 @author: chb69
```
Comment (yuanzhou) [inline hubmap_commons/activity.py:4]: This `activity.py` has been deprecated for a while along with `entity.py`,  `provenance.py`, `metadata.py`, `uuid_generator.py`, and `neo4j_connection.py`. Is this a good time to remove them as part of the refactoring? The `hubmap_const.py` should also ne gone, but it's still being used in the Auth Helper even though the majority of that constants are no longer relevant.

**Review by yuanzhou — CHANGES_REQUESTED** [review]

File: hubmap_commons/hm_auth.py (hunk 1 above)
Comment (shirey) [reply hubmap_commons/hm_auth.py:17]: @yuanzhou Not sure how it got there...removing it now.

File: hubmap_commons/activity.py (hunk 2 above)
Comment (shirey) [reply hubmap_commons/activity.py:4]: @yuanzhou Agree- I'll remove now.

Comment (yuanzhou) [reply hubmap_commons/activity.py]: @shirey you removed `activity.py`, `entity.py`, `provenance.py`, and `metadata.py`  but didn't remove `uuid_generator.py` and `neo4j_connection.py`, are you still using them somewhere?

Comment (shirey) [reply hubmap_commons/activity.py]: @yuanzhou I removed those remaining two files and pushed.

Comment (yuanzhou) [reply hubmap_commons/activity.py]: Turned out that I had to add back all the deleted modules, they are all being used by the ingest-api...  except this globus_groups.py can be refactored out without any concerns [38e8d24](https://github.com/hubmapconsortium/commons/pull/92/commits/38e8d2439af9fd192fd4c2a4b46d8a769a82a5b1)

**Pull request merged**